File: java/src/main/java/com/egjaedong/tinyclaw/provider/LlmProvider.java
package com.egjaedong.tinyclaw.provider;

import java.util.List;

import com.egjaedong.tinyclaw.schema.Message;
import com.egjaedong.tinyclaw.schema.ToolDefinition;

/**
 * 与大模型通信的统一契约。对照 {@code go/internal/provider/interface.go}。
 */
public interface LlmProvider {

    Message generate(List<Message> messages, List<ToolDefinition> availableTools);
}

File: ts/src/provider/llm-provider.ts
import type { Message, ToolDefinition } from "../schema/message";

/** 与大模型通信的统一契约。对照 go/internal/provider/interface.go */
export interface LlmProvider {
  generate(
    messages: Message[],
    availableTools: ToolDefinition[] | null,
  ): Promise<Message>;
}
